xbmc/network/WebServer.cpp

```diff
@@ -1234,6 +1232,45 @@ void CWebServer::UnregisterRequestHandler(IHTTPRequestHandler *handler)
   m_requestHandlers.erase(std::remove(m_requestHandlers.begin(), m_requestHandlers.end(), handler), m_requestHandlers.end());
 }
 
+void CWebServer::LogRequest(HTTPRequest request) const
+{
+  if (!g_advancedSettings.CanLogComponent(LOGWEBSERVER))
+    return;
+
+  std::multimap<std::string, std::string> headerValues;
+  HTTPRequestHandlerUtils::GetRequestHeaderValues(request.connection, MHD_HEADER_KIND, headerValues);
+  std::multimap<std::string, std::string> getValues;
+  HTTPRequestHandlerUtils::GetRequestHeaderValues(request.connection, MHD_GET_ARGUMENT_KIND, getValues);
+
+  CLog::Log(LOGDEBUG, "CWebServer[%hu]  [IN] %s %s %s", m_port, request.version, GetHTTPMethod(request.method).c_str(), request.pathUrlFull.c_str());
```

Comment:
Nice spotting. I'm already too much used to the improved formatting of `StringUtils::Format()`